### PR TITLE
Add even more tasks: token classification, translation and zero shot classification

### PR DESCRIPTION
**Pull request by Wauplin**
Also related to some updates in the specs: https://github.com/huggingface/huggingface.js/pull/872. Related PR is not merged yet but we can still merge this one IMO.

+ everything from https://github.com/huggingface/hub-docs/pull/1399

### CODE REVIEW

File: docs/api-inference/tasks/summarization.md
Comment (osanseviero) [inline docs/api-inference/tasks/summarization.md:24]: This is missing all the parameters of `text2text-generation` no?

File: docs/api-inference/_toctree.yml
Comment (osanseviero) [inline docs/api-inference/_toctree.yml:33]: I'm wondering if we should add an extra layer to split by modality, but not needed I think?

File: docs/api-inference/tasks/token_classification.md
Comment (osanseviero) [inline docs/api-inference/tasks/token_classification.md:33]: Note for myself: many of these are not supported in `-community`

File: docs/api-inference/tasks/token_classification.md
Comment (osanseviero) [inline docs/api-inference/tasks/token_classification.md:78]: Nit I just saw: let's remove the extra newline

File: docs/api-inference/tasks/translation.md
Comment (osanseviero) [inline docs/api-inference/tasks/translation.md:27]: Where do these come from? I'm looking at https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/translation/inference.ts#L20 and https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/tasks/translation/spec/input.json

File: docs/api-inference/tasks/zero_shot_classification.md
Comment (osanseviero) [inline docs/api-inference/tasks/zero_shot_classification.md:25]: note for Omar, in the current API this is managed through parameters, not inputs?

**Review by osanseviero — COMMENTED** [review]
Cool stuff!

File: docs/api-inference/tasks/summarization.md
Comment (Wauplin) [reply docs/api-inference/tasks/summarization.md]: Yes indeed. I've made some updates in https://github.com/huggingface/huggingface.js/pull/872 and it's better now. Parameters have been added in https://github.com/huggingface/hub-docs/pull/1398/commits/825571b3d9610278cc72359d004ddef0bd865987

File: docs/api-inference/_toctree.yml
Comment (Wauplin) [reply docs/api-inference/_toctree.yml:33]: I wondered as well. For now I'm listing everything by alphabetical order. I think it's fine to keep it like this but if you are opinionated we can keep the same "per-modality split" as in hf.co/tasks.

Comment (Wauplin) [reply docs/api-inference/_toctree.yml:33]: (I'm always using ctrl+F so I'm not of good advice for this :see_no_evil:)

File: docs/api-inference/tasks/translation.md
Comment (Wauplin) [reply docs/api-inference/tasks/translation.md:27]: I opened a PR in `@huggingface.js/tasks` to add modifications to the existing specs (see https://github.com/huggingface/huggingface.js/pull/872). And [here](https://github.com/huggingface/hub-docs/blob/4039c7eb276f14139e86d851259074e9631ad11e/scripts/api-inference/scripts/generate.ts#L18) in the `generate.ts` script I make sure to pull the specs from the `"update-specification-for-docs"` revision. We will have to switch to main once all specs are good.

**Comment by Wauplin** [comment]
Thanks for the review @osanseviero! I've addressed all comments and fixed what had to be fixed. Mind doing another quick pass? :pray:

File: docs/api-inference/_toctree.yml
Comment (osanseviero) [reply docs/api-inference/_toctree.yml:33]: Yeah let's wait for final review from others!

File: docs/api-inference/tasks/translation.md
Comment (osanseviero) [inline docs/api-inference/tasks/translation.md:103]: Maybe an example in english given the model is english->french?

**Review by osanseviero — APPROVED** [review]
Thanks!

**Pull request merged**
